diff --git a/Cakefile.js b/Cakefile.js
index d4e733b..d2b0cc8 100644
--- a/Cakefile.js
+++ b/Cakefile.js
@@ -279,6 +279,7 @@ task('bench', 'quick benchmark of compilation time', () => {
 // Run the CoffeeScript test suite.
 var runTests = function (CoffeeScript) {
   CoffeeScript.register();
+  require('babel-register');
   const startTime = Date.now();
   let currentFile = null;
   let passedTests = 0;
@@ -354,14 +355,14 @@ var runTests = function (CoffeeScript) {
   if (!generatorsAreAvailable) { files.splice(files.indexOf('generators.coffee'), 1); }

   for (const file of Array.from(files)) {
-    if (helpers.isCoffee(file)) {
+    if (file.endsWith('.js')) {
       var filename;

       const literate = helpers.isLiterate(file);
       currentFile = (filename = path.join('test', file));
       const code = fs.readFileSync(filename);
       try {
-        CoffeeScript.run(code.toString(), { filename, literate });
+        require(`./${filename}`);
       } catch (error1) {
         const error = error1;
         failures.push({ filename, error });
@@ -372,7 +373,12 @@ var runTests = function (CoffeeScript) {
 };


-task('test', 'run the CoffeeScript language test suite', () => runTests(CoffeeScript));
+task('test', 'run the CoffeeScript language test suite', () => {
+  const success = runTests(CoffeeScript);
+  if (!success) {
+    process.exitCode = 1;
+  }
+});


 task('test:browser', 'run the test suite against the merged browser script', () => {
diff --git a/check-coffeescript-examples.sh b/check-coffeescript-examples.sh
new file mode 100755
index 0000000..7638b14
--- /dev/null
+++ b/check-coffeescript-examples.sh
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+# Simple shell script to compare the decaffeinated CoffeeScript 1.10.0 compiler
+# with the official run, running it on all CoffeeScript files in the repo and
+# making sure the output is the same. For now, it just fails on the first error.
+
+set -e
+
+rm -rf examples-tmp
+mkdir examples-tmp
+cd examples-tmp
+git clone https://github.com/jashkenas/coffeescript.git --branch 1.10.0
+
+for path in $(find coffeescript -name '*.coffee'); do
+  echo "Comparing ${path}..."
+  (./coffeescript/bin/coffee --compile --print ${path} || echo 'ERROR') > expected.js
+  (../bin/coffee --compile --print ${path} || echo 'ERROR') > actual.js
+  cmp expected.js actual.js
+  echo 'Passed!'
+done
diff --git a/test/cluster.js b/test/cluster.js
index 706ca0d..07f9fd6 100644
--- a/test/cluster.js
+++ b/test/cluster.js
@@ -8,6 +8,10 @@

 if (typeof testingBrowser !== 'undefined' && testingBrowser !== null) { return; }

+// CoffeeScript overwrites process.argv, which this test depends on, so manually
+// overwrite it instead.
+process.argv[1] = __filename;
+
 const cluster = require('cluster');

 if (cluster.isMaster) {
