diff --git a/spec/compile-cache-spec.js b/spec/compile-cache-spec.js
index 195938fe2..0bd70ea8d 100644
--- a/spec/compile-cache-spec.js
+++ b/spec/compile-cache-spec.js
@@ -110,7 +110,7 @@ describe('CompileCache', () => {
       waits(1);
       return runs(() => {
         error = new Error('Oops again');
-        expect(error.stack).toContain('compile-cache-spec.coffee');
+        expect(error.stack).toContain('compile-cache-spec.js');
         return expect(Array.isArray(error.getRawStack())).toBe(true);
       });
     });
@@ -122,7 +122,7 @@ describe('CompileCache', () => {
       Error.prepareStackTrace = originalPrepareStackTrace;

       const error = new Error('Oops');
-      expect(error.stack).toContain('compile-cache-spec.coffee');
+      expect(error.stack).toContain('compile-cache-spec.js');
       return expect(Array.isArray(error.getRawStack())).toBe(true);
     });

@@ -135,7 +135,7 @@ describe('CompileCache', () => {
       };

       const error = new Error('Oops');
-      expect(error.stack).toContain('compile-cache-spec.coffee');
+      expect(error.stack).toContain('compile-cache-spec.js');
       expect(error.foo).toBe('bar');
       return expect(Array.isArray(error.getRawStack())).toBe(true);
     });
diff --git a/static/index.html b/static/index.html
index 39c7d80c1..8d2c85de4 100644
--- a/static/index.html
+++ b/static/index.html
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
 <head>
-  <meta http-equiv="Content-Security-Policy" content="default-src *; img-src blob: data: *; script-src 'self'; style-src 'self' 'unsafe-inline'; media-src blob: data: mediastream: *;">
+  <meta http-equiv="Content-Security-Policy" content="default-src *; img-src blob: data: *; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; media-src blob: data: mediastream: *;">
   <script src="index.js"></script>
 </head>
 <body tabindex="-1">
