diff --git a/Cakefile b/Cakefile
index 040b67f..43f8bb8 100644
--- a/Cakefile
+++ b/Cakefile
@@ -1,4 +1,5 @@
 fs = require 'fs'
+child_process = require 'child_process'
 glob = require 'glob'
 path = require 'path'
 browserify = require 'browserify'
@@ -9,8 +10,7 @@ copySync = (src, dest) ->
     fs.writeFileSync dest, fs.readFileSync(src)

 coffeeSync = (input, output) ->
-    coffee = fs.readFileSync(input).toString()
-    fs.writeFileSync output, CoffeeScript.compile(coffee)
+    child_process.execSync("./node_modules/.bin/babel #{input} --out-file #{output}")

 task 'compile', 'Compile Coffeelint', ->
     console.log 'Compiling Coffeelint...'
@@ -19,22 +19,23 @@ task 'compile', 'Compile Coffeelint', ->
     invoke 'compile:commandline'

 task 'compile:commandline', 'Compiles commandline.js', ->
-    coffeeSync 'src/commandline.coffee', 'lib/commandline.js'
-    coffeeSync 'src/configfinder.coffee', 'lib/configfinder.js'
-    coffeeSync 'src/cache.coffee', 'lib/cache.js'
-    coffeeSync 'src/ruleLoader.coffee', 'lib/ruleLoader.js'
+    coffeeSync 'src/commandline.js', 'lib/commandline.js'
+    coffeeSync 'src/configfinder.js', 'lib/configfinder.js'
+    coffeeSync 'src/cache.js', 'lib/cache.js'
+    coffeeSync 'src/ruleLoader.js', 'lib/ruleLoader.js'
     fs.mkdirSync 'lib/reporters' unless fs.existsSync 'lib/reporters'
-    for src in glob.sync('reporters/*.coffee', { cwd: 'src' })
+    for src in glob.sync('reporters/*.js', { cwd: 'src' })
         # Slice the "coffee" extension of the end and replace with js
-        dest = src[...-6] + 'js'
+        dest = src
         coffeeSync "src/#{src}", "lib/#{dest}"

 task 'compile:browserify', 'Uses browserify to compile coffeelint', ->
     opts =
         standalone: 'coffeelint'
     b = browserify(opts)
-    b.add [ './src/coffeelint.coffee' ]
+    b.add [ './src/coffeelint.js' ]
     b.transform require('coffeeify')
+    b.transform require('babelify')
     b.bundle().pipe fs.createWriteStream('lib/coffeelint.js')

 task 'prepublish', 'Prepublish', ->
diff --git a/bin/coffeelint b/bin/coffeelint
index 3148fdc..56c21cd 100755
--- a/bin/coffeelint
+++ b/bin/coffeelint
@@ -19,6 +19,7 @@ var thisdir = path.dirname(fs.realpathSync(__filename));
 // I use this so vim runs the newest code while I work on CoffeeLint. -Asa
 commandline = path.join(thisdir, '..', "lib", "commandline.js");
 if (!existsFn(commandline)) {
+    require('babel-register');
     require('coffee-script/register');
     require('../src/commandline');
 } else {
diff --git a/package.json b/package.json
index 287f170..bbc9419 100644
--- a/package.json
+++ b/package.json
@@ -45,7 +45,7 @@
   "license": "MIT",
   "scripts": {
     "pretest": "cake compile",
-    "test": "./vowsrunner.js --spec test/*.coffee test/*.litcoffee",
+    "test": "./vowsrunner.js --spec test/*.js test/*.litcoffee",
     "testrule": "npm run compile && ./vowsrunner.js --spec",
     "posttest": "npm run lint",
     "prepublish": "cake prepublish",
diff --git a/vowsrunner.js b/vowsrunner.js
index 2433979..b589667 100755
--- a/vowsrunner.js
+++ b/vowsrunner.js
@@ -1,4 +1,5 @@
 #!/usr/bin/env node
+require('babel-register')
 require('coffee-script/register')
 require('./node_modules/vows/bin/vows')

